src/components/Nav/index.tsx

```diff
@@ -92,16 +104,44 @@ const Nav: FC<IProps> = ({ path }) => {
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    transform: "rotate(30deg)", // Rotate the icon to 30 degrees on hover
+                    color: "blue", // Change color to blue on hover
+                  }}
                   onClick={toggleColorMode}
                 ></IconButton>
+
                 <ButtonLink
+                  onMouseOver={() => setLanguagesHover(true)}
+                  onMouseOut={() => setLanguagesHover(false)}
+                  as={Link}
                   to={`/languages/${fromPageParameter}`}
-                  leftIcon={<Icon as={MdLanguage} />}
                   variant="ghost"
                   isSecondary
                   px={1.5}
+                  _hover={{
+                    color: "blue", // Change color to blue on hover
+                  }}
```

Comment:
@pettinarip this is actually a bug, where the `isSecondary` prop is causing an override to any pseudo class. So adding `_hover` is necessary here for now.

@rohan9024 please keep to semantic tokens and refer to the [Ethereum DS Figma file](https://www.figma.com/file/NrNxGjBL0Yl1PrNrOT8G2B/ethereum.org-Design-System?type=design&node-id=2283-45220&mode=design&t=5c6BwHguyB4nMFv1-0) when determining which semantic token to use. Here, the token should be `primary.hover`

```suggestion
                  _hover={{
                    color: "primary.hover", // Change color to blue on hover
                  }}
```